fetcher/fetcher_file.go

```diff
@@ -0,0 +1,82 @@
+package fetcher
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+	"time"
+)
+
+// FILE is used to check new version
+type FILE struct {
+	PATH     string
+	Interval time.Duration
+	// file modify time and its inode number makes up its tag
+	uniqTag string
+	delay   bool
+}
+
+// Init interval and lastTag
+func (f *FILE) Init() error {
+	if f.PATH == "" {
+		return fmt.Errorf("PATH required")
+	}
+	if f.Interval == 0 {
+		f.Interval = 10 * time.Second
+	}
+
+	if err := f.updateTag(); err != nil {
+		return err
+	}
+	f.delay = false
+
+	return nil
+}
+
+// Fetch file
+func (f *FILE) Fetch() (io.Reader, error) {
+	//delay fetches after first
+	if f.delay {
+		time.Sleep(f.Interval)
+	}
+	f.delay = true
+
+	lastTag := f.uniqTag
+	if err := f.updateTag(); err != nil {
+		return nil, err
+	}
+	// no change
+	if lastTag == f.uniqTag {
+		return nil, nil
+	}
+
+	file, err := os.Open(f.PATH)
+	if err != nil {
+		return nil, err
+	}
+
+	return file, nil
+}
+
+func (f *FILE) updateTag() error {
+	file, err := os.Open(f.PATH)
+	if err != nil {
+		// new version not exist, return
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return fmt.Errorf("Open file error: %s", err)
+	}
+	defer file.Close()
+
+	state, err := file.Stat()
+	if err != nil {
+		return fmt.Errorf("Get file state error: %s", err)
+	}
+
+	info := state.Sys().(*syscall.Stat_t)
+	f.uniqTag = fmt.Sprintf("%d%d", info.Ctim.Nano(), info.Ino)
```

Comment:
Hashing isn't too bad, SSDs read the disk at GB/sec so I don't think its a big issue. overseer is [already hashing the binary here](https://github.com/jpillora/overseer/blob/master/proc_master.go#L83-L90). Nevertheless, I think it's safe to use `state.mtime|state.filesize`